packages/react/src/components/TimePicker/TimePicker.test.tsx

```diff
@@ -233,103 +227,81 @@ describe('TimePicker', () => {
       />,
     );
 
-    const timePickerComboBox = getByRole('combobox') as HTMLInputElement;
-    expect(_errorMessage).toEqual('');
+    const combobox = screen.getByRole('combobox');
+    expect(lastError).toBe('');
 
-    userEvent.click(timePickerComboBox);
-    userEvent.type(timePickerComboBox, '11111 AM{enter}');
-    const firstExpectedErrorMessage = 'Enter a valid time in the 24-hour format: hh:mm:ss';
-    expect(_errorMessage).toEqual(firstExpectedErrorMessage);
+    // First invalid
+    userEvent.click(combobox);
+    userEvent.type(combobox, '11111 AM{enter}');
+    const firstError = 'Enter a valid time in the 24-hour format: hh:mm:ss';
+    expect(lastError).toBe(firstError);
     expect(onValidationResult).toHaveBeenCalled();
     onValidationResult.mockClear();
 
-    // verify that onValidationResult is not called twice for the same error message
-    userEvent.clear(timePickerComboBox);
-    userEvent.click(timePickerComboBox);
-    userEvent.type(timePickerComboBox, '88888 AM{enter}');
-    expect(_errorMessage).toEqual(firstExpectedErrorMessage);
+    // Same invalid again => no callback
+    userEvent.clear(combobox);
+    userEvent.click(combobox);
+    userEvent.type(combobox, '88888 AM{enter}');
+    expect(lastError).toBe(firstError);
     expect(onValidationResult).not.toHaveBeenCalled();
 
-    // verify that onValidationResult is finally called again for a new error message
-    userEvent.click(timePickerComboBox);
-    userEvent.type(timePickerComboBox, '03:00:00{enter}');
+    // New error (out of range)
+    userEvent.click(combobox);
+    userEvent.type(combobox, '03:00:00{enter}');
     expect(onValidationResult).toHaveBeenCalled();
-
-    const secondExpectedErrorMessage = 'Please enter a time within the range';
-    expect(_errorMessage).toContain(secondExpectedErrorMessage);
+    const secondError = 'Please enter a time within the range';
+    expect(lastError).toContain(secondError);
   });
 
   describe('validates entered text when', () => {
-    it('receives an invalid hour input for 24-hour-no-seconds format', () => {
-      const wrapper = mount(<TimePicker id="test" allowFreeform useHour12={false} showSeconds={false} />);
-      const input = wrapper.find('input');
-      input.simulate('input', { target: { value: '95:22' } });
-
-      // Trigger validation
-      input.simulate('keydown', { which: KeyCodes.enter });
+    const simulateValidation = async (props: Parameters<typeof render>[0]) => {
```

Comment:
this looks like halucination, can we change the naming and api signature ? 

- the argument it accepts is actually JSXElement not props